aspnet/mvc/models/validation.rst

```diff
@@ -1,8 +1,196 @@
+   
+  <input class="form-control" type="datetime" 
+  data-val="true" 
+  data-val-classicmovie="Classic movies must have a release year earlier than 1960" 
+  data-val-required="The ReleaseDate field is required." 
+  id="ReleaseDate" name="ReleaseDate" value="" />
+
+JQuery's unobtrusive validation uses the data in the ``data-`` attributes to display error messages. However, jQuery doesn't know about rules or messages until you add them to jQuery's ``validator`` object. This is shown in the example below that adds a method named ``classicmovie`` containing custom client validation code to the jQuery ``validator`` object, then rules and errors. 
+
+.. code-block:: javascript
+   
+  $(function () {
+      jQuery.validator.addMethod('classicmovie',
+      function (value, element, params) {
+	      // custom validation code
+          return false;
+      }, '');
+
+      jQuery.validator.unobtrusive.adapters.add('classicmovie',
+      function (options) {
+          options.rules['classicmovie'] = {};
+          options.messages['classicmovie'] = options.message;
+      });
+  }(jQuery));
+
+Now jQuery has the information to execute the custom JavaScript validation as well as the error message to display if that validation code returns false. 
+
+Remote valiation
+----------------
+Remote validation is a great feature to use when you need to validate data on the client against data on the server. For example, your app may need to verify whether an email or user name is already in use, and it must query a large amount of data to do so. Downloading large sets of data for validating one or a few fields consumes too many resources. It may also expose sensitive information. An alternative is to make a round-trip request to validate a field; however, doing so diminishes the user's experience, especially when other fields quickly pass client side validation.
+
+You can implement remote validation in a two step process. First, you must annotate your model with the ``[Remote]`` attribute. The ``[Remote]`` attribute accepts multiple overloads you can use to direct client side JavaScript to the appropriate code to call. The example points to the ``VerifyEmail`` action method of the ``Users`` controller. 
+
+.. literalinclude:: validation/sample/User.cs
+ :language: c#
+ :lines: 9-17
+ :dedent: 4
+ 
+The second step is putting the validation code in the corresponding action method as defined in the ``[Remote]`` attribute. In this case, the code goes in the ``VerifyEmail`` method of the ``UsersController``. It returns a ``JsonResult`` that the client side can use to proceed or pause and display a validation error if needed.
+ 
+.. literalinclude:: validation/sample/UsersController.cs
+ :language: c#
+ :lines: 7-21   
+ :dedent: 4
+
+Now when users enter an email, JavaScript in the view makes a remote call to see if that email has been taken, and if so, then displays the error message. Otherwise, the user can submit the form as usual.  
```

Comment:
> if that email has been taken

Either change the code to perform a database lookup to check if any user object exists with the given username or change this description to match the regular expression enforced. In the second case, add a comment that users should actually use `[RegularExpression]` for such a requirement (and it will be automatically supported client-side).